### PR TITLE
chapter 9

### PR DESCRIPTION
**Pull request by chan227**

### CODE REVIEW

**Comment by chan227** [comment]
ready

File: app/models/user.rb [hunk 1]
```diff
@@ -21,4 +22,22 @@ def User.digest(string)
...
+    SecureRandom.urlsafe_base64
+  end
+
+  def remember
+    self.remember_token = User.new_token
+    update_attribute :remember_digest, User.digest(remember_token)
+  end
+
+  def authenticated? remember_token
+    return false if remember_digest.nil?
```
Comment (vanvtt-0952) [inline app/models/user.rb]: change ` return false if remember_digest.nil?` to ` return unless remember_digest` or ` return false unless remember_digest`

Comment (chan227) [reply app/models/user.rb]: Noted it, thank you.

**Comment by chan227** [comment]
ready

**Comment by vanvtt-0952** [comment]
merged

**Pull request merged**